### PR TITLE
Enriching events

### PR DESCRIPTION
**Pull request by bottaio**

### CODE REVIEW

File: flink-tutorial-base/src/main/java/com/getindata/tutorial/base/input/EnrichedSongsSource.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (grzegorz8) [inline flink-tutorial-base/src/main/java/com/getindata/tutorial/base/input/EnrichedSongsSource.java]: Remove these headers in each file.

File: tasks/task1/src/main/java/com/getindata/FilterSongs.java [hunk 2]
```diff
@@ -1,38 +1,79 @@
...
+    }
+
+    static class SelectInvalidEvents implements FlatMapFunction<Either<SongEvent, EnrichedSongEvent>, SongEvent> {
+        @Override
+        public void flatMap(Either<SongEvent, EnrichedSongEvent> event, Collector<SongEvent> collector) throws Exception {
+            // todo: fill in the code
+        }
+    }
+
+    static class EnrichmentFunction extends RichMapFunction<SongEvent, Either<SongEvent, EnrichedSongEvent>> {
```
Comment (grzegorz8) [inline tasks/task1/src/main/java/com/getindata/FilterSongs.java]: move it before `SelectValidEvents`.

File: tasks/task1/src/test/java/com/getindata/FilterSongsTest.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
     void shouldAcceptRelevantSong() {
```
Comment (grzegorz8) [inline tasks/task1/src/test/java/com/getindata/FilterSongsTest.java]: These tests are completely out-of-date.

File: flink-tutorial-base/src/main/java/com/getindata/tutorial/base/enrichmennt/EnrichmentService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.getindata.tutorial.base.model.Song;
+
+import java.util.Optional;
+
+import static com.getindata.tutorial.base.input.utils.Songs.SONGS;
+
+public class EnrichmentService {
+    /**
+     * Fetches song by it's identifier. If song could not be found - returns empty Optional.
```
Comment (grzegorz8) [inline flink-tutorial-base/src/main/java/com/getindata/tutorial/base/enrichmennt/EnrichmentService.java]: nit: it's -> its

File: flink-tutorial-base/src/main/java/com/getindata/tutorial/base/enrichmennt/EnrichmentService.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+     */
+    public Optional<Song> getSongById(long songId) {
+        for (Song s : SONGS) {
```
Comment (grzegorz8) [inline flink-tutorial-base/src/main/java/com/getindata/tutorial/base/enrichmennt/EnrichmentService.java]: SONGS as a Map?

File: flink-tutorial-base/src/main/java/com/getindata/tutorial/base/kafka/KafkaProperties.java [hunk 6]
```diff
@@ -1,52 +1,17 @@
...
+    public static final String INPUT_TOPIC = "input";
 
-    public static String getUsername() {
-        // FIXME please return your username
-        // return "lion";
-        throw new UnsupportedOperationException("Please provide your user name.");
-    }
+    public static final String INPUT_AVRO_TOPIC = "input-avro";
+    public static final String OUTPUT_AVRO_TOPIC = "output-avro";
+    public static final String SCHEMA_REGISTRY_URL = "http://localhost:8082";
```
Comment (grzegorz8) [inline flink-tutorial-base/src/main/java/com/getindata/tutorial/base/kafka/KafkaProperties.java]: localhost?

File: tasks/task1/src/main/java/com/getindata/EnrichSongs.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                .map(event -> "Could not enrich event: " + event)
+                .print();
+
+        // execute streams
+        sEnv.execute("Example program");
+    }
+
+    static class EnrichmentFunction extends RichMapFunction<SongEvent, Either<SongEvent, EnrichedSongEvent>> {
+        @Override
+        public Either<SongEvent, EnrichedSongEvent> map(SongEvent songEvent) throws Exception {
```
Comment (grzegorz8) [inline tasks/task1/src/main/java/com/getindata/EnrichSongs.java:40]: maybe add a todo explaining that EnrichmentService can be used?

File: tasks/task3/pom.xml [hunk 8]
```diff
@@ -14,5 +14,34 @@
     <artifactId>task3</artifactId>
     <version>1.0-SNAPSHOT</version>
 
-
+    <build>
+        <plugins>
```
Comment (grzegorz8) [inline tasks/task3/pom.xml:18]: These plugins are repeated in every sub-module. Let's define them in `build` or `plugingManagement` in parent.

File: tasks/task3/src/main/avro/SongEventAvro.avsc [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (grzegorz8) [inline tasks/task3/src/main/avro/SongEventAvro.avsc:1]: We should add avro files containing all fields is some subdirectory.

Comment (bottaio) [reply tasks/task3/src/main/avro/SongEventAvro.avsc:1]: they are defined in `flink-tutorial-base` (we need them in generation jobs). if we wanted to provide them in `task3` project, we'd have to put them in some artificial packages (and I don't really like it)

**Review by grzegorz8 — APPROVED** [review]

**Pull request merged**